cmd/util/cmd/execution-state-extract/execution_state_extract.go

```diff
@@ -348,32 +348,42 @@ func createTrieFromPayloads(logger zerolog.Logger, payloads []*ledger.Payload) (
 
 func newMigrations(
 	log zerolog.Logger,
-	dir string,
+	outputDir string,
 	nWorker int, // number of concurrent worker to migation payloads
 	runMigrations bool,
 ) []ledger.Migration {
 	if runMigrations {
-		rwf := reporters.NewReportFileWriterFactory(dir, log)
+		rwf := reporters.NewReportFileWriterFactory(outputDir, log)
+
+		var acctBasedMigrations []migrators.AccountBasedMigration
+
+		if flagFilterUnreferencedSlabs {
+			acctBasedMigrations = append(acctBasedMigrations, migrators.NewFilterUnreferencedSlabsMigration(
+				outputDir,
+				rwf,
+			))
+		}
+
+		acctBasedMigrations = append(acctBasedMigrations, migrators.NewAtreeRegisterMigrator(
+			rwf,
+			flagValidateMigration,
+			flagLogVerboseValidationError,
+			flagContinueMigrationOnValidationError,
+			flagCheckStorageHealthBeforeMigration,
+			flagCheckStorageHealthAfterMigration,
+		))
+
+		acctBasedMigrations = append(acctBasedMigrations, &migrators.DeduplicateContractNamesMigration{})
+
+		// This will fix storage used discrepancies caused by the previous migrations
+		acctBasedMigrations = append(acctBasedMigrations, &migrators.AccountUsageMigrator{})
```

Comment:
Can maybe merge the three migrations, to avoid the repeated 
`acctBasedMigrations = append(acctBasedMigrations, ...`

```suggestion
		acctBasedMigrations = append(
			acctBasedMigrations,
			migrators.NewAtreeRegisterMigrator(
				rwf,
				flagValidateMigration,
				flagLogVerboseValidationError,
				flagContinueMigrationOnValidationError,
				flagCheckStorageHealthBeforeMigration,
				flagCheckStorageHealthAfterMigration,
			),
			&migrators.DeduplicateContractNamesMigration{},

			// This will fix storage used discrepancies caused by the previous migrations
			&migrators.AccountUsageMigrator{},
		)
```